vspec/vssexporters/vss2protobuf.py

```diff
@@ -127,18 +127,41 @@ def traverse_signal_tree(tree: VSSNode, proto_file):
     # write proto messages to file
     for tree_node in filter(lambda n: n.is_branch(), PreOrderIter(tree)):
         proto_file.write(f"message {tree_node.qualified_name('')} {{" + "\n")
-        print_message_body(tree_node.children, proto_file)
+        print_message_body(tree_node.children, proto_file, static_uid, add_optional)
         proto_file.write("}\n\n")
 
 
-def print_message_body(nodes, proto_file):
+def print_message_body(nodes, proto_file, static_uid, add_optional):
+    usedKeys = {}
     for i, node in enumerate(nodes, 1):
-        data_type = node.qualified_name("")
         if not (node.is_branch() or node.is_struct()):
             dt_val = node.get_datatype()
             data_type = mapped.get(dt_val.strip("[]"), dt_val.strip("[]"))
-            data_type = ("repeated " if dt_val.endswith("[]") else "") + data_type
-        proto_file.write(f"  {data_type} {node.name} = {i};" + "\n")
+            if dt_val.endswith("[]"):
+                data_type = "repeated " + data_type
+            elif add_optional:
+                data_type = "optional " + data_type
+        else:
+            data_type = node.qualified_name("")
+            if add_optional:
+                data_type = "optional " + data_type
+        if static_uid:
+            fieldNumber = int(int(node.extended_attributes['staticUID'], 0) / 8)
```

Comment:
Could we possibly try to add a more user friendly message if a node does not have a `staticUID`.

Like first checking if there is a `staticUID` and if not give a user friendly message, like "to use --static-uid option you must first ..." and then exit

```
erik@debian4:~/vehicle_signal_specification$ ./vss-tools/vspec2protobuf.py -u ./spec/units.yaml --strict --add-optional --static-uid ./spec/VehicleSignalSpecification.vspec new.proto
INFO     Added 29 quantities from /home/erik/vehicle_signal_specification/spec/quantities.yaml
INFO     Added 61 units from ./spec/units.yaml
INFO     Loading vspec from ./spec/VehicleSignalSpecification.vspec...
INFO     Calling exporter...
INFO     Generating protobuf output...
Traceback (most recent call last):
  File "/home/erik/vehicle_signal_specification/./vss-tools/vspec2protobuf.py", line 25, in <module>
    vspec2x.main(sys.argv[1:])
  File "/home/erik/vss-tools/vspec/vspec2x.py", line 155, in main
    self.generator.generate(args, tree, self.vspec2vss_config, data_type_tree)
  File "/home/erik/vss-tools/vspec/vssexporters/vss2protobuf.py", line 193, in generate
    traverse_signal_tree(signal_root, f, config.static_uid, config.add_optional)
  File "/home/erik/vss-tools/vspec/vssexporters/vss2protobuf.py", line 130, in traverse_signal_tree
    print_message_body(tree_node.children, proto_file, static_uid, add_optional)
  File "/home/erik/vss-tools/vspec/vssexporters/vss2protobuf.py", line 149, in print_message_body
    fieldNumber = int(int(node.extended_attributes['staticUID'], 0) / 8)
KeyError: 'staticUID'
```

Reply:
Good point, I added a check and logging statement.